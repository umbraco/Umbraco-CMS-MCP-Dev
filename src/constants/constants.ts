export const BLANK_UUID = "00000000-0000-0000-0000-000000000000";
export const ROOT_DOCUMENT_TYPE_ID = "a95360e8-ff04-40b1-8f46-7aa4b5983096";
export const CONTENT_DOCUMENT_TYPE_ID = "b871f83c-2395-4894-be0f-5422c1a71e48";
export const Default_Memeber_TYPE_ID = "d59be02f-1df9-4228-aa1e-01917d806cda";
export const TextString_DATA_TYPE_ID = "0cc0eba1-9960-42c9-bf9b-60e150b429ae";
export const IMAGE_MEDIA_TYPE_ID = "cc07b313-0843-4aa8-bbda-871c8da728c8";
export const FOLDER_MEDIA_TYPE_ID = "f38bd2d7-65d0-48e6-95dc-87ce06ec2d3d";
export const EXAMPLE_IMAGE_PATH =
  "/src/umb-management-api/tools/temporary-file/__tests__/helpers/example.jpg";
