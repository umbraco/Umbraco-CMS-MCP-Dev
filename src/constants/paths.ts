import { join } from 'path';

// Use absolute path from current working directory
export const RESOURCES_DIR = join(process.cwd(), 'build', 'resources').replace(/\\/g, '/');