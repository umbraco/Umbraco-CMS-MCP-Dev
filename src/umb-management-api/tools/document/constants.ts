export const UmbracoDocumentPermissions = {
    Create: "Umb.Document.Create",
    Read: "Umb.Document.Read",
    Update: "Umb.Document.Update",
    Delete: "Umb.Document.Delete",
    Publish: "Umb.Document.Publish",
    Unpublish: "Umb.Document.Unpublish",
    Move: "Umb.Document.Move",
    Sort: "Umb.Document.Sort",
    Duplicate: "Umb.Document.Duplicate",
    PublicAccess: "Umb.Document.PublicAccess",
    CultureAndHostnames: "Umb.Document.CultureAndHostnames",
    CreateBlueprint: "Umb.Document.CreateBlueprint",
}