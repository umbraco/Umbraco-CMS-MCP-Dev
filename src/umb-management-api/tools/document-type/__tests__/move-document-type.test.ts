import MoveDocumentTypeTool from "../put/move-document-type.js";
import { DocumentTypeBuilder } from "./helpers/document-type-builder.js";
import { DocumentTypeFolderBuilder } from "./helpers/document-type-folder-builder.js";
import { DocumentTypeTestHelper } from "./helpers/document-type-test-helper.js";
import { jest } from "@jest/globals";
import { BLANK_UUID } from "@/constants/constants.js";

describe("move-document-type", () => {
  const TEST_DOCTYPE_NAME = "_Test DocumentType Move";
  const TEST_FOLDER_NAME = "_Test Folder";
  let originalConsoleError: typeof console.error;

  beforeEach(() => {
    originalConsoleError = console.error;
    console.error = jest.fn();
  });

  afterEach(async () => {
    console.error = originalConsoleError;
    // Clean up any test document types
    await DocumentTypeTestHelper.cleanup(TEST_DOCTYPE_NAME);
    await DocumentTypeTestHelper.cleanup(TEST_FOLDER_NAME);
  });

  it("should move a document type", async () => {
    // Create a folder
    const folderBuilder = await new DocumentTypeFolderBuilder(
      TEST_FOLDER_NAME
    ).create();

    // Create a document type to move
    const builder = await new DocumentTypeBuilder()
      .withName(TEST_DOCTYPE_NAME)
      .create();

    // Move the document type
    const result = await MoveDocumentTypeTool().handler(
      {
        id: builder.getId(),
        data: {
          target: {
            id: folderBuilder.getId(),
          },
        },
      },
      { signal: new AbortController().signal }
    );

    // Verify the handler response using snapshot
    expect(result).toMatchSnapshot();

    // Verify the document type was moved
    const found = await DocumentTypeTestHelper.findDocumentType(
      TEST_DOCTYPE_NAME
    );
    expect(found).toBeDefined();
    expect(found!.id).toBe(builder.getId());
  });

  it("should handle moving to non-existent folder", async () => {
    // Create a document type to move
    const builder = await new DocumentTypeBuilder()
      .withName(TEST_DOCTYPE_NAME)
      .create();

    const result = await MoveDocumentTypeTool().handler(
      {
        id: builder.getId(),
        data: {
          target: {
            id: BLANK_UUID,
          },
        },
      },
      { signal: new AbortController().signal }
    );

    // Verify the error response using snapshot
    expect(result).toMatchSnapshot();
  });

  it("should handle moving non-existent document type", async () => {
    const result = await MoveDocumentTypeTool().handler(
      {
        id: BLANK_UUID,
        data: {
          target: {
            id: BLANK_UUID,
          },
        },
      },
      { signal: new AbortController().signal }
    );

    // Verify the error response using snapshot
    expect(result).toMatchSnapshot();
  });
});
