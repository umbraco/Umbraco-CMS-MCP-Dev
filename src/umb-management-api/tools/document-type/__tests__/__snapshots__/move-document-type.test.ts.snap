// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`move-document-type should handle moving non-existent document type 1`] = `
{
  "content": [
    {
      "text": "Error using move-document-type:
{
  "message": "Request failed with status code 404",
  "response": {
    "type": "Error",
    "title": "Not Found",
    "status": 404,
    "detail": "The specified document type was not found",
    "operationStatus": "NotFound"
  }
}",
      "type": "text",
    },
  ],
}
`;

exports[`move-document-type should handle moving to non-existent folder 1`] = `
{
  "content": [
    {
      "text": "Error using move-document-type:
{
  "message": "Request failed with status code 404",
  "response": {
    "type": "Error",
    "title": "Container not found",
    "status": 404,
    "detail": "The specified container was not found",
    "operationStatus": "ContainerNotFound"
  }
}",
      "type": "text",
    },
  ],
}
`;

exports[`move-document-type should move a document type 1`] = `
{
  "content": [
    {
      "text": """",
      "type": "text",
    },
  ],
}
`;
